__tests__/integration/template/transaction/template.test.ts

```diff
@@ -0,0 +1,346 @@
+      .build();
+
+    const tx = await interpreter.build(template, DEBUG);
+    await transactionUtils.signTransaction(tx, hWallet.storage, DEFAULT_PIN_CODE);
+    tx.prepareToSend();
+    const sendTx = new SendTransaction({ storage: hWallet.storage, transaction: tx });
+    await sendTx.runFromMining();
+    await waitForTxReceived(hWallet, tx.hash, null);
+
+    expect(tx.outputs[0].tokenData).toBe(0);
+    expect(tx.outputs[0].value).toBe(1n);
+    expect(tx.outputs[1].tokenData).toBe(129);
+    expect(tx.outputs[1].value).toBe(2n);
+  });
+
+  it('should be able to complete a transaction inputs', async () => {
+    const address = await hWallet.getAddressAtIndex(25);
+    const template = new TransactionTemplateBuilder()
+      .addSetVarAction({ name: 'addr', value: address })
+      .addSetVarAction({ name: 'token', value: tokenUid })
+      .addSetVarAction({
+        name: 'tk_balance',
+        call: { method: 'get_wallet_balance', token: '{token}' },
+      })
+      .addTokenOutput({ address: '{addr}', amount: '{tk_balance}', token: '{token}' })
+      .addAuthorityOutput({ address: '{addr}', authority: 'mint', token: '{token}' })
+      .addAuthorityOutput({ address: '{addr}', authority: 'mint', token: '{token}' })
+      .addCompleteAction({})
+      .addShuffleAction({ target: 'all' })
+      .build();
+
+    const tx = await interpreter.build(template, DEBUG);
+    await transactionUtils.signTransaction(tx, hWallet.storage, DEFAULT_PIN_CODE);
+    tx.prepareToSend();
+    const sendTx = new SendTransaction({ storage: hWallet.storage, transaction: tx });
+    await sendTx.runFromMining();
+    await waitForTxReceived(hWallet, tx.hash, null);
+    expect(tx.hash).toBeDefined();
+  });
+
+  it('should be able to complete a transaction change using the complete instruction', async () => {
+    const address = await hWallet.getAddressAtIndex(25);
+    const template = new TransactionTemplateBuilder()
+      .addSetVarAction({ name: 'addr', value: address })
+      .addSetVarAction({ name: 'token', value: tokenUid })
+      .addSetVarAction({
+        name: 'tk_balance',
+        call: { method: 'get_wallet_balance', token: '{token}' },
+      })
+      .addUtxoSelect({ fill: '{tk_balance}', token: '{token}', autoChange: false })
+      .addAuthoritySelect({ token: '{token}', authority: 'mint' })
+      .addAuthoritySelect({ token: '{token}', authority: 'melt' })
+      .addTokenOutput({ address: '{addr}', amount: 1, token: '{token}' })
+      .addCompleteAction({})
+      .build();
+
+    const tx = await interpreter.build(template, DEBUG);
+    await transactionUtils.signTransaction(tx, hWallet.storage, DEFAULT_PIN_CODE);
+    tx.prepareToSend();
+    const sendTx = new SendTransaction({ storage: hWallet.storage, transaction: tx });
+    await sendTx.runFromMining();
+    await waitForTxReceived(hWallet, tx.hash, null);
+    expect(tx.hash).toBeDefined();
```

Comment:
suggestion: validate that there are 4 outputs

Reply:
Done